### PR TITLE
Add CodeQL workflow

### LINKED ISSUE #8137: LGTM does not support Java 17
**Description**

LGTM does not support Java 17: https://lgtm.com/help/lgtm/java-extraction
This blocks us in upgrading modules to language level 17. There are multiple ways to go forward:

* Halt migration to Java 17.
* Disable LGTM temporarily
* Migrate to CodeQL

The first two options are suboptimal because LGTM appears to not be actively maintained anymore.

Migrating to [CodeQL](https://codeql.github.com) would be ideal because it offers superior integration with GitHub, for example issues found by CodeQL show up in the Security tab in GitHub. Integrating CodeQL seems fairly simple, there is an official action: https://github.com/github/codeql-action

relates to #8096

### PR DESCRIPTION
**Pull request by lenaschoenburg**
## Description

Adds a CodeQL workflow that should be similar to LGTM. I've tested introducing an issue similar to this one: https://lgtm.com/projects/g/camunda-cloud/zeebe/rev/pr-ac3b6a67cd862f9923402ba727a0b39234d73f39
but for CodeQL to work properly it needs to run on the develop branch first, so this PR just adds the CodeQL workflow without removing LGTM.

## Related issues

relates to #8137 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/0.25`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

**Comment by oleschoenburg** [comment]
Also of note here: LGTM considers more issues to be deal-breakers for a PR. For example this one: https://lgtm.com/rules/2049510531/ has severity "recommendation" but for CodeQL we have configured to only break builds if an issue with severity "high" is found. We might want to adjust that.
Screenshot 2021-11-04 at 18 04 55

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+          distribution: 'temurin'
+          java-version: '17'
+          cache: 'maven'
+      - name: Initialize CodeQL
+        uses: github/codeql-action/init@v1
+        with:
+          languages: java
+          queries: +security-and-quality
+      - name: Build
+        run: mvn -T1C -DskipTests -DskipChecks install
```
Comment (npepinpe) [inline .github/workflows/codeql.yml:27]:
```suggestion
        run: mvn -B -T1C -DskipTests -DskipChecks install
```

We probably want to run this in batch mode (i.e. non interactive)

**Review by npepinpe — APPROVED** [review]
:+1: 

So we'll have another PR to check if it catches similar issues?

File: .github/workflows/codeql.yml (hunk 1 above)
Comment (oleschoenburg) [reply .github/workflows/codeql.yml:27]: Thanks, didn't know about this one!

**Comment by oleschoenburg** [comment]
> Regarding the severity, this is adjustable at anytime, no?

Yes, I don't see why not.

**Comment by oleschoenburg** [comment]
> So we'll have another PR to check if it catches similar issues?

Yes, I'll open one tomorrow that checks that it catches similar issues and then remove LGTM

**Comment by oleschoenburg** [comment]
bors merge

**Pull request merged**